integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -80,160 +74,139 @@ spec:
       {{- end}}
       containers:
         - name: alluxio-master
-          image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
-          imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+          image: {{ .Values.image }}:{{ .Values.imageTag }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
           securityContext:
-            runAsUser: {{ $root.Values.user }}
-            runAsGroup: {{ $root.Values.group }}
-          {{- if $root.Values.master.resources  }}
-          resources:
-            {{- if $root.Values.master.resources.limits }}
-            limits:
-              cpu: {{ $root.Values.master.resources.limits.cpu }}
-              memory: {{ $root.Values.master.resources.limits.memory }}
-            {{- end }}
-            {{- if $root.Values.master.resources.requests }}
-            requests:
-              cpu: {{ $root.Values.master.resources.requests.cpu }}
-              memory: {{ $root.Values.master.resources.requests.memory }}
-            {{- end }}
+            runAsUser: {{ .Values.user }}
+            runAsGroup: {{ .Values.group }}
+          {{- if .Values.master.resources  }}
+{{ include "alluxio.master.resources" . | indent 10 }}
           {{- end }}
           command: ["/entrypoint.sh"]
-          {{- if $root.Values.master.args }}
+          {{- if .Values.master.args }}
           args:
-{{ toYaml $root.Values.master.args | trim | indent 12 }}
+{{ toYaml .Values.master.args | trim | indent 12 }}
           {{- end }}
           {{- if $isHaEmbedded }}
           env:
+          - name: POD_HOSTNAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
           - name: ALLUXIO_MASTER_JAVA_OPTS
-            value: {{ $masterJavaOpts | quote }}
+            value: " -Dalluxio.master.hostname=${POD_HOSTNAME} "
```

Comment:
not sure I understand this change. 
1. is the pod hostname the same as the service name? can you give an example of the hostname for the embedded configuration?
2. what about other master java opts in addition to `alluxio.master.hostname`. are those lost?

Reply:
Now the workflow has changed a bit. With only one StatefulSet and multiple replicas, there's no longer one Service that controls the whole StatefulSet. So the serviceName field in StatefulSet doesn't really do anything (but this is a compulsory field).

Now I have to manually create one NodePort Service for each Pod in the StatefulSet, with service name equals to Pod name. That's why the master hostname has changed from StatefulSet's serviceName to Pod name (which equals to the service name).

Reply:
For Point 2 this is a great question. I need to double check the hostname-related java opts.

Reply:
for 2, my question is that will the hostname property be appended to ALLUXIO_MASTER_JAVA_OPTS from the configMap